cli/azd/pkg/project/service_config.go

```diff
@@ -38,8 +40,10 @@ type ServiceConfig struct {
 	Docker DockerProjectOptions `yaml:"docker"`
 	// The infrastructure provisioning configuration
 	Infra provisioning.Options `yaml:"infra"`
+	// Hook configuration for service
+	Hooks map[string]*ext.HookConfig `yaml:"hooks,omitempty"`
 
-	handlers map[Event][]ServiceLifecycleEventHandlerFn
+	*ext.EventDispatcher[ServiceLifecycleEventArgs] `yaml:",omitempty"`
```

Comment:
Happy to take this separately, but the event dispatcher seems more likely that it should be put inside the business logic classes - `Service` and `FrameworkService`, rather than the config object.

Reply:
Lets take this as a separate issue.  It was originally added to the `Config` objects because it is used for provisioned instances as well.  Actual `Service` instances assume you already have target resources provisioned and available.